partiql-eval/src/main/kotlin/org/partiql/eval/internal/helpers/ValueUtility.kt

```diff
@@ -0,0 +1,97 @@
+     * already of the expected type, or it will be a null value of the expected type.
+     */
+    @OptIn(PartiQLValueExperimental::class)
+    fun PQLValue.check(type: PartiQLValueType): PQLValue {
+        if (this.type == type) {
+            return this
+        }
+        if (!this.isNull) {
+            throw TypeCheckException("Expected type $type but received ${this.type}.")
+        }
+        return PQLValue.nullValue(type)
+    }
+
+    /**
+     * Returns the underlying string value of a PartiQL text value
+     *
+     * @throws NullPointerException if the value is null
+     * @throws TypeCheckException if the value's type is not a text type (string, symbol, char)
+     */
+    @OptIn(PartiQLValueExperimental::class)
+    fun PQLValue.getText(): String {
+        return when (this.type) {
+            PartiQLValueType.STRING -> this.stringValue
+            PartiQLValueType.SYMBOL -> this.symbolValue
+            PartiQLValueType.CHAR -> this.charValue
+            else -> throw TypeCheckException("Expected text, but received ${this.type}.")
+        }
+    }
+
+    /**
+     * Takes in a [PQLValue] that is any integer type ([PartiQLValueType.INT8], [PartiQLValueType.INT8],
+     * [PartiQLValueType.INT8], [PartiQLValueType.INT8], [PartiQLValueType.INT8]) and returns the [BigInteger] (potentially
+     * coerced) that represents the integer.
+     *
+     * @throws NullPointerException if the value is null
+     * @throws TypeCheckException if type is not an integer type
+     */
+    @OptIn(PartiQLValueExperimental::class)
+    fun PQLValue.getBigIntCoerced(): BigInteger {
+        return when (this.type) {
+            PartiQLValueType.INT8 -> this.int8Value.toInt().toBigInteger()
+            PartiQLValueType.INT16 -> this.int16Value.toInt().toBigInteger()
+            PartiQLValueType.INT32 -> this.int32Value.toBigInteger()
+            PartiQLValueType.INT64 -> this.int64Value.toBigInteger()
+            PartiQLValueType.INT -> this.intValue
+            else -> throw TypeCheckException()
+        }
+    }
+
+    /**
+     * Takes in a [PQLValue] that is any integer type ([PartiQLValueType.INT8], [PartiQLValueType.INT8],
+     * [PartiQLValueType.INT8], [PartiQLValueType.INT8], [PartiQLValueType.INT8]) and returns the [Int] (potentially
+     * coerced) that represents the integer.
+     *
+     * @throws NullPointerException if the value is null
+     * @throws TypeCheckException if type is not an integer type
+     */
+    @OptIn(PartiQLValueExperimental::class)
+    fun PQLValue.getInt32Coerced(): Int {
+        return when (this.type) {
+            PartiQLValueType.INT8 -> this.int8Value.toInt()
+            PartiQLValueType.INT16 -> this.int16Value.toInt()
+            PartiQLValueType.INT32 -> this.int32Value
+            PartiQLValueType.INT64 -> this.int64Value.toInt()
+            PartiQLValueType.INT -> this.intValue.toInt()
+            else -> throw TypeCheckException()
+        }
+    }
```

Comment:
I suggest adding all of these coerces to the PQLValue implementations.

```java
class Int32Value ... {

    private Int value;

    @Override
    short toShort() { 
       // throw DataException on overflow  etc.
      return value as short;
     }

    @Override
    int getInt() { return value };

    @Override
    long getLong() { return value };
}

Reply:
See https://github.com/partiql/partiql-lang-kotlin/pull/1451/commits/7f345443ec0d3479fcba47f1931eb0fec3370ef2#diff-7273fc22dadb098a56a8f9e14afb134d7fdce01af8709eff841f91576bee9d3e